src/physics/fluxsurfaces.jl

```diff
@@ -1613,18 +1613,24 @@ function flux_surfaces(eqt::equilibrium__time_slice{T1}, wall_r::AbstractVector{
     # first find a guess for the starting point
     #   by finding all interior extrema that could be near axes
     #   then excluding points outside the first wall (i.e., likely coils)
-    pts = (psi_sign > 0.0) ? IMASutils.findall_interior_argmin(PSI) : IMASutils.findall_interior_argmax(PSI)
-    wall = collect(zip(wall_r, wall_z))
-    pts_in_wall = [PolygonOps.inpolygon((r[i], z[j]), wall) for (i, j) in pts] # 1 if in wall
-    npts_in_wall = sum(pts_in_wall)
-    if npts_in_wall == 0
-        error("Could not find guess for magnetic axis inside wall")
-    elseif npts_in_wall > 1
-        error("Multiple guesses for magnetic axis found inside wall")
-    end
-    ig, jg = pts[argmax(pts_in_wall)] # this are the indices for the guess inside the wall
-    rguess, zguess = r[ig], z[jg]
-    RA, ZA = find_magnetic_axis(r, z, PSI_interpolant, psi_sign; rguess, zguess)
+    if !isempty(wall_r)
```

Comment:
Thank you for doing this @bclyons12 ! 👍 

I wonder if we should add this logic to the `find_magnetic_axis()` function.

Reply:
You would have to pass the wall to it then, so up to you. You could make a separate function that takes the wall as an argument and does this I suppose.